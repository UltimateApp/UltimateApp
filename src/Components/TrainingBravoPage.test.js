import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { connect, Provider } from 'react-redux';
import { render, fireEvent, cleanup } from 'react-native-testing-library';

import fixtures from '../Fixtures/TestFixtures';
import store from '../Store/testStore';

import ConnectedTrainingBravoPage, { TrainingBravoPage } from './TrainingBravoPage';

afterEach(cleanup);
beforeEach(() => jest.useFakeTimers()); // for Animated
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<TrainingBravoPage />', () => {
  const navigation = { navigate: jest.fn(), setOptions: jest.fn() };
  const program = fixtures.programs[0];
  const training = program.trainings[0];
  const drill = training.drills[0];
  const route = {
    params: {
      program,
      training,
      drill,
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<TrainingBravoPage route={route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('marks training as complete when finished', async () => {
    const trainingLastDrill = training.drills[training.drills.length - 1];
    const completeTraining = jest.fn();

    const Stack = createStackNavigator();
    const navigate = jest.fn();

    const MockedConnectedTrainingBravoPage = connect(null, () => ({ completeTraining }))(TrainingBravoPage);

    const { getByTestId } = render(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              name="TrainingBravoPage"
              component={MockedConnectedTrainingBravoPage}
              initialParams={{ program, training, drill: trainingLastDrill }}
              listeners={({ navigation }) => ({
                transitionStart: e => {
                  navigation.navigate = navigate;
                },
              })}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>,
    );

    await fireEvent.press(getByTestId('button'));

    expect(completeTraining).toBeCalledWith({ training, program });
    expect(navigate).toBeCalledWith('ProgramsPage');
  });
});
