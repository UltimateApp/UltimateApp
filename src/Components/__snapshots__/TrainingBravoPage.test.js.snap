// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingBravoPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 20,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
          "paddingHorizontal": 20,
        }
      }
    >
      Beginner
       - session 
      1
      /
      2
    </Text>
    <Text>
      Well done! You've just finished the training!
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "500",
        "marginBottom": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    Recap
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingHorizontal": 20,
      }
    }
  >
    You worked on:
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "paddingHorizontal": 20,
      }
    }
  >
    - 
    Legs
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "paddingHorizontal": 20,
      }
    }
  >
    - 
    Defense
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 20,
        "flex": 1,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 20,
        }
      }
    >
      Get ready for the next training
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#232b2b",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
          "width": "80%",
        }
      }
      testID="button"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Finish Training!
      </Text>
    </View>
  </View>
</View>
`;
